### DISCUSSION TITLE
Error Compiling

### DISCUSSION THREAD

**Discussion by ranganath36** [iotivity/iotivity-lite#297] [General]
Hello,

   I am trying to compile the Iotivity Windows code using MS VS 17. I am getting the following errors and I have no clue how to resolve this. Request advice on the below errors.

1. Error	MSB3073	The command "if not exist "..\deps\mbedtls\patched.txt" C:\Iotivity\..\patch-mbedtls.bat
:VCEnd" exited with code 9009.	IoTivity-lite	C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets	203	

2. Error	MSB3073	The command "if exist ...\x64\Debug\onboarding_tool_creds rmdir  C:\Iotivity\\x64\Debug\onboarding_tool_creds /s /q
if exist C:\Iotivity\\x64\Debug\pki_certs rmdir  C:\Iotivity\\x64\Debug\pki_certs /s /q
xcopy /s /y /d C:\Iotivity\..\..\..\apps\pki_certs\rootca*.pem C:\Iotivity\\x64\Debug\pki_certs\
:VCEnd" exited with code 4.	onboarding_tool	C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets	123	

Thanks and Regards.

**Comment by Danielius1922** [comment]
I wasn't even aware that we have MSVS projects in the repository. When I'm running MSVS2022 on my Win machine I just open the iotivity-lite root folder and it uses the CMake it finds there. Won't that work for you?

@jkralik do we want to keep this folder - https://github.com/iotivity/iotivity-lite/tree/master/port/windows/vs2015 ? I guess we must for now, because the SWIG stuff hasn't been ported to CMake yet.

**Reply by ranganath36** [reply]
Hello,

   Thank you for the reply. Is there a way to compile on MSVS 2017?

Thanks and Regards

**Reply by Danielius1922** [reply]
As far as I know, MSVS 2017 has CMake integrated so just open the root folder of your iotivity-lite checkout (where the main CMakeLists.txt file is). The when you select Build / Build All (Ctrl + Shift + B) then the default targets defined in CMake for Windows MSVC should be build.

If that doesn't work for you, I've tried fixing the MSVS solutions - checkout this branch [adam/bugfix/msvc-solution](https://github.com/iotivity/iotivity-lite/tree/adam/bugfix/msvc-solution) and see if it works. But be warned that eventually we will get rid of everything that is not CMake (Makefiles on Unixes, MSVS solutions), because CMake itself can generate various build systems.

**Reply by ranganath36** [reply]
Hello,

   Thank you once again. With the code provided in the link, I am still facing an issue. Below is the error I am getting..

  fatal: Not a git repository (or any of the parent directories): .git
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets(203,5): error MSB3073: The command "if not exist "...\iotivity-lite-adam-bugfix-msvc-solution\port\windows\vs2015\..\..\..\deps\mbedtls\patched.txt" ...\iotivity-lite-adam-bugfix-msvc-solution\port\windows\vs2015\..\patch-mbedtls.bat
C:\Program Files (x86)\MSBuild\Microsoft.Cpp\v4.0\V140\Microsoft.CppCommon.targets(203,5): error MSB3073: :VCEnd" exited with code 1.

Please suggest how to overcome this.

Thanks and Regards

**Reply by Danielius1922** [reply]
Before compilation the patch-mbedtls.bat batch file is executed and seems that it fails on your machine. Please do the following steps and  copy the output you get:

* open Command Prompt
* navigate in your checkout of iotivity-lite to port\windows directory in the Command Prompt
* execute patch-mbedtls.bat
